eclass/apache-module.eclass

```diff
@@ -44,14 +44,19 @@
 # </IfDefine>
 # @CODE
 
-case ${EAPI} in
-	6|7|8) ;;
-	*) die "${ECLASS}: EAPI ${EAPI:-0} not supported" ;;
-esac
 
 if [[ -z ${_APACHE_MODULE_ECLASS} ]]; then
 _APACHE_MODULE_ECLASS=1
 
+case ${EAPI} in
+        6)
+		ewarn "${CATEGORY}/${PF}: ebuild uses ${ECLASS} with deprecated EAPI ${EAPI}!"
+		ewarn "${CATEGORY}/${PF}: Support will be removed in 30 days. Please port to newer EAPI."
```

Comment:
Technically, these should be `eqawarn` because their purpose is to inform ebuild authors, not users.

I can't say that I like that noise in global scope. Just die and be done with it. EAPI 6 has been deprecated for more than three years, so overlay maintainers had plenty of time to update.

Reply:
"fatal error: EAPI 6 not supported" is certainly not an eqawarn!

The purpose of this is absolutely to inform users, not authors. eqawarn is something we should have done a year ago. At this late stage, we need to be informing *users* that the packages they rely on are about to cease availability unless they go encourage the ebuild author to do something about it.

Reply:
Installed packages won't magically disappear from users' systems; they just won't get any updates for these packages. Then again, I very much doubt that there are updates for any package that is stuck at EAPI 6.

Reply:
> Installed packages won't magically disappear from users' systems; they just won't get any updates for these packages. 

This is especially problematic when it comes to subslot rebuilds where users often get very confusing conflicts. Anyway, I won't relitigate the reasons for doing this again.

Reply:
> Installed packages won't magically disappear from users' systems; they just won't get any updates for these packages. Then again, I very much doubt that there are updates for any package that is stuck at EAPI 6.

They don't have to get updates, they just have to be able to be reinstalled after a subslot rebuild...

> I disagree that we should have done eqawarn earlier. It would have caused noise when sourcing ebuilds in the Gentoo repo, e.g. during metadata generation. IMHO eclasses are the wrong place for these warnings, and for future EAPI deprecations it should be a task for the package manager. (For example. it could read eapis-banned from layout.conf and warn when trying to install an ebuild with a matching EAPI.)

I agree! We work with the technology we currently have, which unfortunately isn't a lot. We should prepare for EAPI 7 deprecation by adding a portage feature to handle it precisely as you say.

This was even mentioned in IRC before.